### PR TITLE
Find peaks

### LINKED ISSUE #2: [Feature] Find Peaks
Finding peaks can be useful to identify common patterns in signals and for segmentation of datasets among many other applications.

https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.find_peaks.html?highlight=peaks#scipy.signal.find_peaks

### PR DESCRIPTION
**Pull request by spectre-ns**
PR to implement a basic implementation for peak finding looking to resolve #2.

### CODE REVIEW

**Comment by spectre-ns** [comment]
@JohanMabille  I know that xtensor requires C++14 or higher. Can this library be C++17? It makes all the metatemplate code so much easier.

**Comment by JohanMabille** [comment]
Yes, we will upgrade xtensor to C++17 at some point.

**Comment by JohanMabille** [comment]
The CI failure is unrelated, I need ot fix it;

**Comment by spectre-ns** [comment]
@JohanMabille thanks! I will need the CI as my build environment is MSVC only and I know there are extensions in the MSVC compiler clang and gcc don't like.

**Comment by JohanMabille** [comment]
The issue comes from the xtensor feedstock, it should be fixed within an hour.

**Comment by JohanMabille** [comment]
Ok there were other issues, so I had to push other commits. Sorry for the conflicts, you need to rebase, but at least the CI is green on main.

**Comment by spectre-ns** [comment]
@JohanMabille Reviewed and addressed the conflicts.

**Comment by JohanMabille** [comment]
Usually we try to avoid merging the main branch in order to keep a clean history. We prefer to rebase, that also allows to squash a bit and keep meaningful commits only (no need to do the work again though, just saying for the next PRs).

**Comment by spectre-ns** [comment]
Apologies... that's what github asked me to do. Noted for next time!

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,7 +31,7 @@ message(STATUS "xtensor-signal v${${PROJECT_NAME}_VERSION}")
 # Dependencies
 # ============
 
-set(xsimd_REQUIRED_VERSION 9.0.1)
+set(xsimd_REQUIRED_VERSION 8.1.0)
```
Comment (JohanMabille) [inline CMakeLists.txt]: You should keep the 9.0.1 dependency here.

Comment (spectre-ns) [reply CMakeLists.txt]: I couldn't get xsimd 9.0.0 on conda or vcpkg. Suggestions for sources?

Comment (JohanMabille) [reply CMakeLists.txt]: Weird, xsimd 9.0.1 is available on conda-forge for windows. What package you get if you run `conda env create -f environment-dev.yml` ?

Comment (spectre-ns) [reply CMakeLists.txt]: I was running in WSL 2 ands got 8.1.0. Vcpkg is trying to update xsimd to 9.1.0 using the following PR: https://github.com/microsoft/vcpkg/pull/26501. I'm waiting for them to make their update so I can develop in windows using my normal workflow. They seem to be having trouble with apache-arrow...

Comment (spectre-ns) [reply CMakeLists.txt]: Fixed this!

**Pull request closed without merging**